### PR TITLE
[move-vm] Log stacktrace for natives

### PR DESCRIPTION
**Pull request by runtian-zhou**
### Description

Fixing #7224 

### Test Plan
Ran the test and see following:
```
┌── test_create_and_transfer ──────
│ error[E11001]: test failure
│     ┌─ /config/workspace/aptos-core/aptos-move/framework/aptos-token-objects/../aptos-framework/sources/object.move:207:16
│     │
│ 207 │     native fun exists_at<T: key>(object: address): bool;
│     │                ^^^^^^^^^
│     │                │
│     │                Test was not expected to error, but it aborted with code 10 originating in the module 0000000000000000000000000000000000000000000000000000000000000001::object rooted here
│     │                In this function in 0x1::object
│ 
│ 
│ stack trace
│       token::create_named_token(/config/workspace/aptos-core/aptos-move/framework/aptos-token-objects/sources/token.move:91)
│       token::create_token_helper(/config/workspace/aptos-core/aptos-move/framework/aptos-token-objects/sources/token.move:601-608)
│       token::test_create_and_transfer(/config/workspace/aptos-core/aptos-move/framework/aptos-token-objects/sources/token.move:342)
│ 
└──────────────────
```

### CODE REVIEW

File: third_party/move/move-vm/runtime/src/interpreter.rs [hunk 1]
```diff
@@ -859,6 +865,7 @@ impl Interpreter {
                 )
             })
             .collect();
+        println!("{:?}", stack_trace);
```
Comment (rahxephon89) [inline third_party/move/move-vm/runtime/src/interpreter.rs:868]: Should this `println!` be added to `get_internal_state()` instead of here?

Comment (runtian-zhou) [reply third_party/move/move-vm/runtime/src/interpreter.rs:868]: Sry I need to get rid of this...

**Review by rahxephon89 — APPROVED** [review]
A linter error needs to be fixed. Other than that, it looks good to me.

**Comment by brmataptos** [comment]
Can we add a test case or two?  Stack of size zero, 1, more.

Probably can do it in move-vm/transactional-tests/tests/native_functions/

**Review by brmataptos — CHANGES_REQUESTED** [review]
Need a few test cases, as I mentioned.

File: third_party/move/move-vm/runtime/src/interpreter.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (brmataptos) [reply third_party/move/move-vm/runtime/src/interpreter.rs]: I note that to help with adding tests, you could add a new native, perhaps vm_error(major_code: u64) as shown in the patch here:
[vm_error.patch.txt](https://github.com/aptos-labs/aptos-core/files/12600318/vm_error.patch.txt)

[the remark this answers is not in the document]
Comment (brmataptos) [reply third_party/move/move-vm/runtime/src/interpreter.rs]: I guess there is also some .move code needed in the corresponding debug.move file, but that's straightforward.

**Comment by runtian-zhou** [comment]
Test case added.

**Review by brmataptos — APPROVED** [review]
Your test could use a few more comments to serve as a useful guide for anyone trying to understand the VM or write more tests in the future, but it's fine.

**Pull request merged**